frontend/model/state.js

```diff
@@ -535,78 +534,65 @@ const getters = {
+        //       His profile picture can be used as the picture of the direct message
+        //       possibly with the badge of the number of partners.
+        const lastJoinedPartner = partners[partners.length - 1]
+        currentGroupDirectMessages[chatRoomId] = {
+          ...getters.ourDirectMessages[chatRoomId],
+          users,
+          partners,
+          lastJoinedPartner,
+          title: partners.map(un => getters.ourContactProfiles[un].displayName || un).join(', '),
+          picture: getters.ourContactProfiles[lastJoinedPartner]?.picture
         }
       }
     }
-    return privateDMs
+    return currentGroupDirectMessages
   },
-  ourGroupDirectMessages (state, getters) {
-    const groupDMs = {}
-    for (const cID of Object.keys(getters.ourDirectMessages)) {
-      if (getters.ourDirectMessages[cID].privacyLevel === CHATROOM_PRIVACY_LEVEL.GROUP && state[cID]) {
-        groupDMs[cID] = getters.ourDirectMessages[cID]
+  // NOTE: this getter is used to find the ID of the direct message in the current group
+  //       with the name[s] of partner[s]. Normally it's more useful to find direct message
+  //       by the partners instead of contractID
+  ourGroupDirectMessageFromUsernames (state, getters) {
+    return (partners) => { // NOTE: string | string[]
+      if (typeof partners === 'string') {
+        partners = [partners]
       }
+      return Object.keys(getters.ourGroupDirectMessages).find(chatRoomId => {
+        const cPartners = getters.ourGroupDirectMessages[chatRoomId].partners
+        return cPartners.length === partners.length && union(cPartners, partners).length === partners.length
+      })
     }
-    return groupDMs
   },
   isDirectMessage (state, getters) {
     // NOTE: identity contract could not be synced at the time of calling this getter
     return chatRoomId => {
       const contractID = chatRoomId || getters.currentChatRoomId
-      return getters.isJoinedChatRoom(contractID) && !!getters.ourDirectMessages[contractID]
-    }
-  },
-  isPrivateDirectMessage (state, getters) {
-    // NOTE: identity contract could not be synced at the time of calling this getter
-    return chatRoomId => {
-      const contractID = chatRoomId || getters.currentChatRoomId
-      return getters.ourDirectMessages[contractID]?.privacyLevel === CHATROOM_PRIVACY_LEVEL.PRIVATE
-    }
-  },
-  isGroupDirectMessage (state, getters) {
-    return chatRoomId => {
-      const contractID = chatRoomId || getters.currentChatRoomId
-      return getters.ourDirectMessages[contractID]?.privacyLevel === CHATROOM_PRIVACY_LEVEL.GROUP
-    }
-  },
-  isPrivateChatRoom (state, getters) {
-    return (chatRoomId: string) => {
-      const contractID = chatRoomId || getters.currentChatRoomId
-      return state[contractID]?.attributes?.privacyLevel === CHATROOM_PRIVACY_LEVEL.PRIVATE
+      return getters.isJoinedChatRoom(contractID) && !!getters.ourGroupDirectMessages[contractID]
```

Comment:
@corrideat This getter, `isDirectMessage`, is used in a variety of places to see if the chatroom is a direct message. He needs some boolean logic here to answer that question, right?

Although, I do agree that the first check (`getters.isJoinedChatRoom(contractID)`) might not be necessary because I *think* that check is already handled by `!!getters.ourGroupDirectMessages[contractID]`:

```js
      // NOTE: skip DMs whose chatroom contracts are not synced yet
      if (!state[chatRoomId] || !state[chatRoomId].users?.[getters.ourUsername]) {
        continue
      }
```

What do you think @Silver-IT?

Reply:
@taoeffect, that's correct. I will simplify this getter as you suggested.